homeassistant/components/group.py

```diff
@@ -179,16 +262,70 @@ def async_setup(hass, config):
     )
 
     @asyncio.coroutine
-    def reload_service_handler(service_call):
+    def reload_service_handler(service):
         """Remove all groups and load new ones from config."""
         conf = yield from component.async_prepare_reload()
         if conf is None:
             return
         yield from _async_process_config(hass, conf, component)
 
+    hass.services.async_register(
+        DOMAIN, SERVICE_RELOAD, reload_service_handler,
+        descriptions[DOMAIN][SERVICE_RELOAD], schema=RELOAD_SERVICE_SCHEMA)
+
+    @asyncio.coroutine
+    def groups_service_handler(service):
+        """Handle dynamic group service functions."""
+        object_id = slugify(service.data[ATTR_NAME])
+
+        if service.service == SERVICE_CREATE:
+            if object_id in service_groups:
+                _LOGGER.warning("Group '%s' already exists!", object_id)
+                return
+
+            new_group = yield from Group.async_create_group(
+                hass, service.data[ATTR_NAME],
+                object_id=object_id,
+                entity_ids=service.data.get(ATTR_ENTITIES),
+                user_defined=service.data.get(ATTR_USER_DEFINED),
+                icon=service.data.get(ATTR_ICON),
+                view=service.data.get(ATTR_VIEW),
+                control=service.data.get(ATTR_CONTROL)
+            )
+
+            service_groups[object_id] = new_group
+            return
+
+        if object_id not in service_groups:
+            _LOGGER.warning("Group '%s' not exists!", object_id)
+            return
+
+        if service.service == SERVICE_DELETE:
+            del_group = service_groups.pop(object_id)
+            yield from del_group.async_stop()
+
+        if service.service == SERVICE_UPDATE_TRACKED_ENTITIES:
+            yield from service_groups[object_id].\
+                async_update_tracked_entity_ids(service.data[ATTR_ENTITIES])
+
+    hass.services.async_register(
+        DOMAIN, SERVICE_CREATE, groups_service_handler,
+        descriptions[DOMAIN][SERVICE_CREATE], schema=CREATE_SERVICE_SCHEMA)
+
+    hass.services.async_register(
+        DOMAIN, SERVICE_DELETE, groups_service_handler,
+        descriptions[DOMAIN][SERVICE_DELETE], schema=DELETE_SERVICE_SCHEMA)
+
+    hass.services.async_register(
+        DOMAIN, SERVICE_UPDATE_TRACKED_ENTITIES, groups_service_handler,
+        descriptions[DOMAIN][SERVICE_UPDATE_TRACKED_ENTITIES],
+        schema=UPDATE_TRACKED_ENTITIES_SERVICE_SCHEMA)
+
     @asyncio.coroutine
     def visibility_service_handler(service):
         """Change visibility of a group."""
+        _LOGGER.warning("'group.set_visibility' is dedicated and will "
```

Comment:
We can easily allow it if we make the `set` service either create or update a group.